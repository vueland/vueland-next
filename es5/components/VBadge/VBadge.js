"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VBadge = void 0;

require("../../../src/components/VBadge/VBadge.scss");

var _vue = require("vue");

var _usePosition = require("../../effects/use-position");

var _useColors2 = require("../../effects/use-colors");

var _useToggle2 = require("../../effects/use-toggle");

var _useElevation2 = require("../../effects/use-elevation");

var _useTransition = require("../../effects/use-transition");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var VBadge = (0, _vue.defineComponent)({
  name: 'v-badge',
  props: _objectSpread(_objectSpread({
    dot: Boolean,
    avatar: Boolean,
    border: Boolean,
    toggle: Boolean,
    content: {
      required: false
    },
    color: {
      type: String,
      "default": 'primary'
    },
    transition: {
      type: String,
      "default": 'scaleIn'
    }
  }, (0, _usePosition.positionProps)()), (0, _useElevation2.elevationProps)()),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;

    var _useElevation = (0, _useElevation2.useElevation)(props),
        elevationClasses = _useElevation.elevationClasses;

    var _useColors = (0, _useColors2.useColors)(),
        setBackground = _useColors.setBackground;

    var offset = (0, _vue.computed)(function () {
      return props.dot ? 4 : 12;
    });

    var calcPosition = function calcPosition() {
      var offsetVal = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
      var value = -offset.value - Number(offsetVal);
      return "".concat(value, "px");
    };

    var computedLeft = (0, _vue.computed)(function () {
      return props.left && calcPosition(props.offsetX);
    });
    var computedRight = (0, _vue.computed)(function () {
      return props.right && calcPosition(props.offsetX);
    });
    var computedTop = (0, _vue.computed)(function () {
      return props.top && calcPosition(props.offsetY);
    });
    var computedBottom = (0, _vue.computed)(function () {
      return props.bottom && calcPosition(props.offsetY);
    });
    var classes = (0, _vue.computed)(function () {
      return {
        'v-badge': true,
        'v-badge--border': props.border,
        'v-badge--dot': props.dot,
        'v-badge--avatar': props.avatar
      };
    });
    var badgeClasses = (0, _vue.computed)(function () {
      return _objectSpread({
        'v-badge__badge': true
      }, elevationClasses.value);
    });
    var styles = (0, _vue.computed)(function () {
      return {
        top: computedTop.value,
        right: computedRight.value,
        bottom: computedBottom.value,
        left: computedLeft.value
      };
    });

    function addContent() {
      if (props.dot) return undefined;
      if (props.content) return String(props.content);
      return undefined;
    }

    function genBadgeSlot() {
      var propsData = {
        "class": {
          'v-badge__badge-slot': true
        }
      };
      return slots.badge ? (0, _vue.h)('div', propsData, slots.badge()) : null;
    }

    function genContent() {
      var propsData = {
        "class": {
          'v-badge__content': true
        }
      };
      return (0, _vue.h)('div', propsData, [addContent(), genBadgeSlot()]);
    }

    function genBadge() {
      var propsData = setBackground(props.color, {
        "class": badgeClasses.value,
        style: styles.value
      });
      return (0, _vue.h)('div', propsData, genContent());
    }

    return function () {
      var badge = genBadge();

      if (props.toggle && !slots.badge) {
        var _useToggle = (0, _useToggle2.useToggle)(props, 'content'),
            isActive = _useToggle.isActive;

        badge = (0, _vue.withDirectives)(badge, [[_vue.vShow, isActive.value]]);
      }

      var propsData = {
        "class": classes.value
      };
      var children = [(0, _useTransition.useTransition)(badge, props.transition), slots["default"] && slots["default"]()];
      return (0, _vue.h)('div', propsData, children);
    };
  }
});
exports.VBadge = VBadge;
//# sourceMappingURL=VBadge.js.map