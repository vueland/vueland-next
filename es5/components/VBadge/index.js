"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VBadge = require("./VBadge");

Object.keys(_VBadge).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _VBadge[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _VBadge[key];
    }
  });
});
//# sourceMappingURL=index.js.map