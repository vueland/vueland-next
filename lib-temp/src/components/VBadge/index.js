export * from './VBadge';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQmFkZ2UvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsY0FBYyxVQUFVLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgKiBmcm9tICcuL1ZCYWRnZSdcbiJdfQ==