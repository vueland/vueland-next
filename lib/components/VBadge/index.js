export * from './VBadge';
//# sourceMappingURL=index.js.map