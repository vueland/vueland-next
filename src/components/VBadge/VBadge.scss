@import "../../styles/scss/modules/global";

.v-badge {
  display: inline-block;
  position: relative;

  &__badge {
    @include flex(flex, center, center);
    border-radius: 50px;
    position: absolute;
    z-index: 10;
    font-size: 10px;
    font-weight: 700;
    min-width: 26px;
    min-height: 26px;
    text-align: center;
    overflow: hidden;
    will-change: transform;
  }

  &__content {
    display: inherit;
    margin: auto;
    padding: 3px;
  }

  &__badge-slot {
    & > * {
      width: 100%;
      min-height: 100%;
      object-fit: contain;
      color: map-get($white, 'base');
    }
  }

  &.v-badge--avatar > .v-badge__badge {
    position: relative;
    width: 90px;
    min-width: 90px;
    height: 90px;
    z-index: 8;
  }

  &.v-badge--dot > .v-badge__badge {
    min-width: .5rem;
    min-height: .5rem;
    padding: 2px;
  }

  &.v-badge--border > .v-badge__badge {
    border-color: map-get($white, 'base') !important;
    border-width: .18rem;
    border-style: solid;
  }
}
