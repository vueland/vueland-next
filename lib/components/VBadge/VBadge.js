import "../../../src/components/VBadge/VBadge.scss";
import { vShow, h, computed, withDirectives, defineComponent } from "vue";
import { positionProps } from "../../effects/use-position";
import { useColors } from "../../effects/use-colors";
import { useToggle } from "../../effects/use-toggle";
import { useElevation, elevationProps } from "../../effects/use-elevation";
import { useTransition } from "../../effects/use-transition";
export const VBadge = defineComponent({
  name: "v-badge",
  props: {
    dot: Boolean,
    avatar: Boolean,
    border: Boolean,
    toggle: Boolean,
    content: {
      required: false
    },
    color: {
      type: String,
      default: "primary"
    },
    transition: {
      type: String,
      default: "scaleIn"
    },
    ...positionProps(),
    ...elevationProps()
  },

  setup(props, {
    slots
  }) {
    const {
      elevationClasses
    } = useElevation(props);
    const {
      setBackground
    } = useColors();
    const offset = computed(() => {
      return props.dot ? 4 : 12;
    });

    const calcPosition = (offsetVal = 0) => {
      const value = -offset.value - Number(offsetVal);
      return `${value}px`;
    };

    const computedLeft = computed(() => {
      return props.left && calcPosition(props.offsetX);
    });
    const computedRight = computed(() => {
      return props.right && calcPosition(props.offsetX);
    });
    const computedTop = computed(() => {
      return props.top && calcPosition(props.offsetY);
    });
    const computedBottom = computed(() => {
      return props.bottom && calcPosition(props.offsetY);
    });
    const classes = computed(() => ({
      "v-badge": true,
      "v-badge--border": props.border,
      "v-badge--dot": props.dot,
      "v-badge--avatar": props.avatar
    }));
    const badgeClasses = computed(() => {
      return {
        "v-badge__badge": true,
        ...elevationClasses.value
      };
    });
    const styles = computed(() => {
      return {
        top: computedTop.value,
        right: computedRight.value,
        bottom: computedBottom.value,
        left: computedLeft.value
      };
    });

    function addContent() {
      if (props.dot) return undefined;
      if (props.content) return String(props.content);
      return undefined;
    }

    function genBadgeSlot() {
      const propsData = {
        class: {
          "v-badge__badge-slot": true
        }
      };
      return slots.badge ? h("div", propsData, slots.badge()) : null;
    }

    function genContent() {
      const propsData = {
        class: {
          "v-badge__content": true
        }
      };
      return h("div", propsData, [addContent(), genBadgeSlot()]);
    }

    function genBadge() {
      const propsData = setBackground(props.color, {
        class: badgeClasses.value,
        style: styles.value
      });
      return h("div", propsData, genContent());
    }

    return () => {
      let badge = genBadge();

      if (props.toggle && !slots.badge) {
        const {
          isActive
        } = useToggle(props, "content");
        badge = withDirectives(badge, [[vShow, isActive.value]]);
      }

      const propsData = {
        class: classes.value
      };
      const children = [useTransition(badge, props.transition), slots.default && slots.default()];
      return h("div", propsData, children);
    };
  }

});
//# sourceMappingURL=VBadge.js.map