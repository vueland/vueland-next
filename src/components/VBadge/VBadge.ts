import './VBadge.scss'

// Vue API
import { vShow, h, computed, withDirectives, defineComponent } from 'vue'

// Effects
import { positionProps } from '../../effects/use-position'
import { useColors } from '../../effects/use-colors'
import { useToggle } from '../../effects/use-toggle'
import { useElevation, elevationProps } from '../../effects/use-elevation'
import { useTransition } from '../../effects/use-transition'

// Types
import { VNode } from 'vue'

export const VBadge = defineComponent({
  name: 'v-badge',

  props: {
    dot: Boolean,
    avatar: Boolean,
    border: Boolean,
    toggle: Boolean,
    content: {
      required: false,
    },
    color: {
      type: String,
      default: 'primary',
    },
    transition: {
      type: String,
      default: 'scaleIn',
    },
    ...positionProps(),
    ...elevationProps(),
  } as any,

  setup(props, { slots }): () => VNode {
    const { elevationClasses } = useElevation(props)
    const { setBackground } = useColors()

    const offset = computed<number>(() => {
      return props.dot ? 4 : 12
    })

    const calcPosition = (offsetVal: string | number = 0): string => {
      const value = -offset.value - Number(offsetVal)
      return `${ value }px`
    }

    const computedLeft = computed<string | boolean>(() => {
      return props.left && calcPosition(props.offsetX)
    })

    const computedRight = computed<string | boolean>(() => {
      return props.right && calcPosition(props.offsetX)
    })

    const computedTop = computed<string | boolean>(() => {
      return props.top && calcPosition(props.offsetY)
    })

    const computedBottom = computed<string | boolean>(() => {
      return props.bottom && calcPosition(props.offsetY)
    })

    const classes = computed<Record<string, boolean>>(() => ({
      'v-badge': true,
      'v-badge--border': props.border,
      'v-badge--dot': props.dot,
      'v-badge--avatar': props.avatar,
    }))

    const badgeClasses = computed<Record<string, boolean>>(() => {
      return {
        'v-badge__badge': true,
        ...elevationClasses.value,
      }
    })

    const styles = computed<Record<string, string | boolean>>(() => {
      return {
        top: computedTop.value,
        right: computedRight.value,
        bottom: computedBottom.value,
        left: computedLeft.value,
      }
    })

    function addContent(): string | undefined {
      if (props.dot) return undefined
      if (props.content) return String(props.content)

      return undefined
    }

    function genBadgeSlot(): VNode | null {
      const propsData = {
        class: {
          'v-badge__badge-slot': true,
        },
      }

      return slots.badge ? h('div',
        propsData, slots.badge()) : null
    }

    function genContent(): VNode {
      const propsData = {
        class: {
          'v-badge__content': true,
        },
      }

      return h('div', propsData, [addContent(), genBadgeSlot()])
    }

    function genBadge(): VNode {
      const propsData = setBackground(
        props.color, {
          class: badgeClasses.value,
          style: styles.value,
        },
      )

      return h('div', propsData, genContent())
    }

    return () => {
      let badge = genBadge()

      if (props.toggle && !slots.badge) {
        const { isActive } = useToggle(props, 'content')
        badge = withDirectives(badge, [[vShow, isActive.value]])
      }

      const propsData = {
        class: classes.value,
      }

      const children = [
        useTransition(badge, props.transition),
        slots.default && slots.default(),
      ]

      return h('div', propsData, children)
    }
  },
})
