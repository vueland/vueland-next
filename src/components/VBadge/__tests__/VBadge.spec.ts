import { h } from 'vue'
import { mount, VueWrapper } from '@vue/test-utils'
import { VBadge } from '../VBadge'

describe('VBadge', () => {
  let mountFunction: (options?: any) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => mount(VBadge, { ...options })
  })

  it('should mount component and match snapshot', () => {
    const cmp = mountFunction()

    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set badge slot and match snapshot', () => {
    const slots = {
      badge: () => 'content',
    }
    const cmp = mountFunction({ slots })

    expect(cmp.find('.v-badge__badge-slot').text()).toEqual(slots.badge())
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set content and match snapshot', () => {
    const props = {
      content: '10',
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').text()).toEqual(props.content)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set color and match snapshot', () => {
    const props = {
      color: 'red darken-3',
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().class).toContain(
      props.color,
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set border and match snapshot', () => {
    const props = {
      border: true,
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge').attributes().class).toContain('v-badge--border')
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set dot style and match snapshot', () => {
    const props = {
      dot: true,
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge').attributes().class).toContain('v-badge--dot')
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set avatar style and match snapshot', () => {
    const props = {
      avatar: true,
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge').attributes().class).toContain('v-badge--avatar')
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should render default slot and match snapshot', () => {
    const slots = {
      default: () => h('div', 'slot content'),
    }
    const cmp = mountFunction({ slots })

    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set left to true and match snapshot', () => {
    const slots = {
      default: () => h('div', 'slot content'),
    }
    const props = {
      left: true,
      content: '10',
    }
    const cmp = mountFunction({ props, slots })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'left: -12px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set right to true and match snapshot', () => {
    const props = {
      right: true,
      content: '10',
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'right: -12px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set top to true and match snapshot', () => {
    const props = {
      top: true,
      content: '10',
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'top: -12px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set bottom to true and match snapshot', () => {
    const props = {
      bottom: true,
      content: '10',
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'bottom: -12px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set offset-x and match snapshot', () => {
    const props = {
      right: true,
      content: '10',
      offsetX: 12,
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'right: -24px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })

  it('should set offset-y and match snapshot', () => {
    const props = {
      top: true,
      content: '10',
      offsetY: -12,
    }
    const cmp = mountFunction({ props })

    expect(cmp.find('.v-badge__badge').attributes().style).toContain(
      'top: 0px;',
    )
    expect(cmp.html()).toMatchSnapshot()
  })
})
