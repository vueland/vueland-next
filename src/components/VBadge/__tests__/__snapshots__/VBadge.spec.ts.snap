// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBadge should mount component and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should render default slot and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
  <div>
    slot content
  </div>
</div>
`;

exports[`VBadge should set avatar style and match snapshot 1`] = `
<div class="v-badge v-badge--avatar">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set badge slot and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
        <div class="v-badge__badge-slot">
          content
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set border and match snapshot 1`] = `
<div class="v-badge v-badge--border">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set bottom to true and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="bottom: -12px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set color and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge red darken-3">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set content and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set dot style and match snapshot 1`] = `
<div class="v-badge v-badge--dot">
  <transition-stub>
    <div class="v-badge__badge primary">
      <div class="v-badge__content">
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set left to true and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="left: -12px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
  <div>
    slot content
  </div>
</div>
`;

exports[`VBadge should set offset-x and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="right: -24px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set offset-y and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="top: 0px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set right to true and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="right: -12px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`VBadge should set top to true and match snapshot 1`] = `
<div class="v-badge">
  <transition-stub>
    <div class="v-badge__badge primary"
         style="top: -12px;"
    >
      <div class="v-badge__content">
        10
      </div>
    </div>
  </transition-stub>
</div>
`;
